Lib/subprocess.py

```diff
@@ -809,6 +822,41 @@ def __init__(self, args, bufsize=-1, executable=None,
             else:
                 line_buffering = False
 
+        gids = []
+        if groups is not None:
+            if isinstance(groups, str):
+                gids = [grp.getgrnam(groups).gr_gid]
+            elif isinstance(groups, int):
+                gids = [groups]
+            elif isinstance(groups, list):
```

Comment:
rather than requiring a `list`, accept a `collections.Sequence` or even `collections.Iterable` as the only thing we require of it is that it can be iterated over.  But we don't really even need this type check once the individual str and int cases are removed.  just let the for loop _be_ the check.  if it isn't iterable, that will fail.

Reply:
I will update so the loop is the check directly.